h5bp/web_performance/cache-control.conf

```diff
@@ -0,0 +1,33 @@
+# ----------------------------------------------------------------------
+# | Cache Control                                                      |
+# ----------------------------------------------------------------------
+
+# TBD
+#
+# (1) TBD
+#
+# (2) TBD
+#
+# (3) TBD
+#
+# https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
+
+map $sent_http_content_type $cache_control {
+  # (1)
+  default           "private, must-revalidate";
+
+  # (2)
+  ~*text/html       "public, immutable";
+  ~*text/css        "public, immutable";
+  ~*text/javascript "public, immutable";
+  ~*xml             "public, immutable";
+  ~*application/pdf "public, immutable";
+
+  # (3)
+  ~*audio/          "public, immutable";
+  ~*image/          "public, immutable";
```

Comment:
I think audio/image/video/pdf should probably not have public, but should use the default behaviour (i.e. no header if we can convince nginx to do that, or maybe an empty header if not).

The documentation seems to discourage using public everywhere. The default behaviour is basically public if no Authorization header is present. These seem like resources which could be restricted by an Authorization header.

Reply:
> The default behaviour is basically public if no Authorization header is present.

Yes! 🙂
But as H5BP is manly focused on static files, Authorization header is almost never expected.
Not 100% sure yet, but I'm considering making public as default.
And as for media (audio/image/video/pdf) I really don't see why it cannot be public, especially where CDNs have a significant impact here.

Reply:
I think documentation again. Looking at the README, it just says it will help improve performance of your server. It doesn't seem to mention what assumptions are being made about your setup.

I'm imagining creating an application, with a file being served from the application, not from the static path served by nginx. This would then be modifying the response from the application to declare it publicly cacheable. That's probably a surprising behaviour if not documented.

Reply:
[See my answer below](https://github.com/h5bp/server-configs-nginx/pull/319#issuecomment-1312926970).